collection_mutation.cc

```diff
@@ -249,22 +249,19 @@ collection_mutation collection_mutation_view_description::serialize(const abstra
     return serialize_collection_mutation(type, tomb, boost::make_iterator_range(cells.begin(), cells.end()));
 }
 
-collection_mutation merge(const abstract_type& type, collection_mutation_view a, collection_mutation_view b) {
-    return a.with_deserialized(type, [&] (collection_mutation_view_description a_view) {
-    return b.with_deserialized(type, [&] (collection_mutation_view_description b_view) {
-
-    assert(type.is_collection());
-    auto& ctype = static_cast<const collection_type_impl&>(type);
-
+template <typename F>
+static collection_mutation_view_description merge(collection_mutation_view_description a, collection_mutation_view_description b, F key_compare) {
```

Comment:
Please add a concept for F as a way to document it (and to provide early type checking)

Reply:
Will use short_type instead